src/kicker/main.c

```diff
@@ -10,15 +10,18 @@
 
 #define NO_COMMAND 0
 
-#define TIMING_CONSTANT 125
+#define TIMING_CONSTANT 10
 #define VOLTAGE_READ_DELAY_MS 40
 
+// 1 ms / 80 us = 12.5
+#define TIMER_TO_MS 12.5f
```

Comment:
If we're using an 8Mhz with 8 scale and a 10 timing constant. shouldn't this be 8e6/8/10 = 1e5 timer period = 
1e-5 timer period. 1 ms = 100 timer.  Also shouldn't this be called TIMER_TO_MS?

Not sure, i think thats rightish. If it is that timing constant seems too low.

Reply:
So the old math also didn't make sense. I think there's an implicit pre-timer prescale of 8 (8Mhz->1MHz) for TIMER0. Then for the counter, we set the timer prescale to 8 (125KHz). We used to count to 125, giving a callback of 1KHz, or every ms. By dropping the count to 10, we get 80 us callbacks. 80us * 12.5 = 1ms. basically by doing this we've just allowed fractions of a ms resolution rather than ms resolution. 

As per the above formula, I think the name is okay. It's ms = timer_val * TIMER_TO_MS

Reply:
Well. In that case, the way you're using it doesn't make sense.

float time_cnt_flt = ((strength / 255.0f) * 9.0f + 0.8f) * TIMER_TO_MS;
You're doing
timer = ms * TIMER_TO_MS

I assume. I think the float is in MS?
=((( I miss the c++ chrono library. static checking for time conversion plzzz

 I didn't see anywhere where the TIMER has an implicit prescale already. I think.
Just going off random blog posts tho. http://maxembedded.com/2011/06/avr-timers-timer0/

What did we setup the attiny clock to be? It has a couple settings from 1 MHz to 8 right? maybe. Embedded so complicatedddd